### ISSUE TITLE
Cannot run the server in DEBUG mode (gradle libertyDebug)

### ISSUE THREAD

**Issue by AndreiPadu** [OpenLiberty/ci.gradle#327]
Recently, I have been trying to start the server in debug mode and it throws the following exception:

**Execution failed for task ':libertyDebug'.

CWWKM2004E: When installDir is set, it must point to a directory that contains lib/ws-launch.jar.**

After several changes, it got me to the following:

**Execution failed for task ':libertyDebug'.
> CWWKM2002E: Failed to invoke [.../build/wlp/bin/server, debug, sgd-serviceServer]. RC= 2 but expected=[0].**

Stacktrace does not show anything helpful either.

The sample provided does not work for me as well, nor other projects. Also, I could not find anything in the documentation to make it work.

TIA!

Note: I am using gradle 5.5.1 and liberty-gradle-plugin:2.6.5.

**Comment by scottkurz** [comment]
This could happen if you haven't installed Liberty first, through the `installLiberty` task.

If you're following the samples, you could start by doing a `gradle build` first then try the **libertyDebug**.  Have you seen this [guide](https://openliberty.io/guides/gradle-intro.html#configure-your-project) walking through a Gradle project with some examples?

**Comment by AndreiPadu** [comment]
Yes, I followed the guide step by step. Running the `installLiberty` task allows me to run the `libertyDebug` task successfully, but it does not start the server. It just builds successfully. Or is there other way to run the server in debug mode like maven's equivalent: `mvn clean package liberty:debug-server`.

**Comment by mattbsox** [comment]
Hi @AndreiPadu, can you provide the list of commands you're calling to start the server with `libertyDebug`? 

You'll need to make sure that the entire build lifecycle has executed before calling `libertyDebug`. Unlike `libertyRun` we haven't specified any task dependencies on `libertyDebug` so there might be some normal build steps that have been missed when you try to run the task. If your build hasn't called `installLiberty`, `libertyCreate`, or `installApps` then you won't be able to run your application in debug mode.

The easiest way to get around this would be to add `libertyDebug.dependsOn "installApps"` to your `build.gradle` file.

**Comment by AndreiPadu** [comment]
Hi @mattbsox. When I call ‘gradle build’, the entire cycle executes. Somehow, there was no **wlp** folder inside the **build** directory. Most likely, I was not executing the entire cycle, like you specified. After I figured that out, I noticed that port 7777 was always busy. When i checked that out in terminal, there was no server running. I had to kill the process manually and everything worked fine after that.
